lib/types/types.go

```diff
@@ -233,3 +227,56 @@ func (d NullDuration) ValueOrZero() Duration {
 
 	return d.Duration
 }
+
+func getInt64(v interface{}) (int64, error) {
+	switch n := v.(type) {
+	case int:
+		return int64(n), nil
+	case int8:
+		return int64(n), nil
+	case int16:
+		return int64(n), nil
+	case int32:
+		return int64(n), nil
+	case int64:
+		return n, nil
+	case uint:
+		return int64(n), nil
+	case uint8:
+		return int64(n), nil
+	case uint16:
+		return int64(n), nil
+	case uint32:
+		return int64(n), nil
+	case uint64:
+		if n > math.MaxInt64 {
+			return 0, fmt.Errorf("%d is too big", n)
+		}
+		return int64(n), nil
+	default:
+		return 0, fmt.Errorf("unable to use type %T as a duration value", v)
+	}
+}
+
+// GetDurationValue is a helper function that can convert a lot of different
+// types to time.Duration.
+//
+// TODO: move to a separate package and check for integer overflows?
+func GetDurationValue(v interface{}) (time.Duration, error) {
+	switch d := v.(type) {
+	case time.Duration:
+		return d, nil
+	case string:
+		return ParseExtendedDuration(d)
+	case float32:
+		return time.Duration(float64(d) * float64(time.Millisecond)), nil
+	case float64:
+		return time.Duration(d * float64(time.Millisecond)), nil
+	default:
+		n, err := getInt64(v)
```

Comment:
Are any beside time.Duration, string, float64 and int64 ... even possible? Because the previous code didn't handle the rest, and I have the feeling that was because we will never actually get one ?

Reply:
I added them out of an abundance of caution... I have no idea if some weird webpack minification magic or something like that won't try to encode plain integer values as something that goja will export as uint8 or whatever... Stranger things have happened... :sweat_smile: And given that this helper function is intentionally outside of `js/`, I also think it makes some sense to handle the other Go int native types. This way the function is more complete and can basically convert every possible value we can recognize as a `time.Duration` to that, no need to second-guess.